spec/models/dialog_group_spec.rb

```diff
@@ -24,19 +24,23 @@
   describe '#update_dialog_fields' do
     let(:dialog_fields) { FactoryGirl.create_list(:dialog_field, 2) }
     let(:dialog_group) { FactoryGirl.create(:dialog_group, :dialog_fields => dialog_fields) }
+    let(:resource_action) { FactoryGirl.create(:resource_action) }
 
     context 'a collection of dialog fields containing two objects with ids and one without an id' do
       let(:updated_fields) do
         [
-          { 'id' => dialog_fields.first.id, 'label' => 'updated_field_label' },
-          { 'id' => dialog_fields.last.id, 'label' => 'updated_field_label' },
+          { 'id' => dialog_fields.first.id, 'label' => 'updated_field_label', 'resource_action' =>
+            {'resource_type' => 'DialogField', 'ae_attributes' => {}} },
+          { 'id' => dialog_fields.last.id, 'label' => 'updated_field_label', 'resource_action' =>
+            {'id' => resource_action.id, 'resource_type' => 'DialogField'} },
           { 'name' => 'new field', 'label' => 'new field label' }
         ]
       end
       it 'creates or updates the dialog fields' do
```

Comment:
@jntullo - One small question - Would it make sense to validate that `dialog_group.update_dialog_fields` will still work as expected if the `resource_action` key is not in the `:updated_fields` (or a different one) hash?